micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/TimeWindowMax.java

```diff
@@ -49,7 +50,8 @@ public class TimeWindowMax {
 
     @SuppressWarnings("ConstantConditions")
     public TimeWindowMax(Clock clock, DistributionStatisticConfig config) {
-        this(clock, config.getExpiry().toMillis(), config.getBufferLength());
+        this(clock, Objects.requireNonNull(config.getExpiry()).toMillis(),
+                Objects.requireNonNull(config.getBufferLength()));
```

Comment:
It feels a bit silly to sprinkle these `Objects.requireNonNull` calls everywhere, which will throw a NPE like would have been thrown without it... I guess it makes it more clear from reading the code where that could happen, at least, but the result is the same.